### ISSUE TITLE
Auth error:Error: read ECONNRESET

### ISSUE THREAD

**Issue by sahas-** [GoogleCloudPlatform/google-cloud-node#1718]
I'm using 0.42.0 and upon (consistent) high volume pub/sub, I keep getting this error from both app engine and local.

```
Auth error:Error: read ECONNRESET
E1017 22:14:00.474261000 140735185842176 plugin_credentials.c:67] Getting metadata from plugin failed with error: read ECONNRESET
E1017 22:14:00.474294000 140735185842176 client_auth_filter.c:93] Client side authentication failure: read ECONNRESET
Error: read ECONNRESET at ../node_modules/google-cloud/node_modules**/grpc/src/node/src/client.js:417:17**
```

In the below snapshot, this happens at the peak

<img width="547" alt="screen shot 2016-10-18 at 7 26 53 am" src="https://cloud.githubusercontent.com/assets/10361685/19477553/5e41fc00-9504-11e6-801b-5bf967c58be2.png">

**Comment by stephenplusplus** [comment]
Is there a way you can provide some code that reproduces this consistently?

**Comment by sahas-** [comment]
Sorry, its internal code. However, I'm happy to share screens and demo you live, if that'll help.

**Comment by stephenplusplus** [comment]
No problem. If possible, removing anything sensitive from your code and reducing it to the simplest possible scenario that reproduces the error will help me debug more quickly. But if that's not possible, I'll need to know what types of API calls are being made and how frequently, so I can try to throw some code together that reproduces.

**Comment by sahas-** [comment]
I'll try to create a sample code to replicate this.
In the mean time,
My app is subscribed to a topic where messages are getting published. Essentially there are 2 different types of messages arrive, they are identified by attributes by the subscriber code to handle them in different code path. Otherwise, both subscriber and publishing applications belongs to the same application and leverage the same service account. 
In short,  what I observed is that this happens on high concurrency of both **pub and sub (~10-15)/sec on the same topic by the same application**.

**Comment by stephenplusplus** [comment]
@sahas- I'm going to close, but please let me know if you can stand up some quick sample code that reproduces this. Generally speaking, there are a couple ways to handle similar scenarios that might relate;

1) throttling calls to the API using a library like [`async`](http://gitnpm.com/async), and/or
2) setting a higher `maxRetries` option on the API calls you make through our library:
    ```js
require('@google-cloud/pubsub')({ maxRetries: 5 })
```
The default value of 3 might not be enough to handle errors that *could* be caused by making a large amount of simultaneous requests.

Hopefully that is useful. If not, re-opening and figuring this specific issue out is only a failing test case away!

**Comment by JustinBeckwith** [comment]
Greetings @weatherbell!  If you're running into issues with Datastore, would you mind opening a new issue here? 
https://github.com/googleapis/nodejs-datastore
